contracts/NativeMinting/L2SyncPoolContracts/L2ScrollSyncPoolETHUpgradeable.sol

```diff
@@ -0,0 +1,125 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.20;
+
+import {
+    MessagingFee,
+    MessagingReceipt
+} from "@layerzerolabs/lz-evm-protocol-v2/contracts/interfaces/ILayerZeroEndpointV2.sol";
+
+import {BaseMessengerUpgradeable} from "../../../utils/BaseMessengerUpgradeable.sol";
+import {BaseReceiverUpgradeable} from "../../../utils/BaseReceiverUpgradeable.sol";
+import {L2BaseSyncPoolUpgradeable} from "../L2BaseSyncPoolUpgradeable.sol";
+import {IL2ScrollMessenger} from "../../../interfaces/IL2ScrollMessenger.sol";
+import {Constants} from "../../../libraries/Constants.sol";
+import {IL1Receiver} from "../../../interfaces/IL1Receiver.sol";
+
+/**
+ * @title L2 Scroll Stack Sync Pool for ETH
+ * @dev A sync pool that only supports ETH on Scroll Stack L2s
+ * This contract allows to send ETH from L2 to L1 during the sync process
+ */
+contract L2ScrollSyncPoolETHUpgradeable is L2BaseSyncPoolUpgradeable, BaseMessengerUpgradeable, BaseReceiverUpgradeable {
+
+    event DepositWithReferral(address indexed sender, uint256 amount, address referral);
+
+    error L2ScrollStackSyncPoolETH__OnlyETH();
+
+    /**
+     * @dev Constructor for L2 Scroll Stack Sync Pool for ETH
+     * @param endpoint Address of the LayerZero endpoint
+     */
+    constructor(address endpoint) L2BaseSyncPoolUpgradeable(endpoint) {
+        _disableInitializers();
+    }
+
+    /**
+     * @dev Initialize the contract
+     * @param l2ExchangeRateProvider Address of the exchange rate provider
+     * @param rateLimiter Address of the rate limiter
+     * @param tokenOut Address of the token to mint on Layer 2
+     * @param dstEid Destination endpoint ID (most of the time, the Layer 1 endpoint ID)
+     * @param messenger Address of the messenger contract (most of the time, the L2 native bridge address)
+     * @param receiver Address of the receiver contract (most of the time, the L1 receiver contract)
+     * @param delegate Address of the owner
+     */
+    function initialize(
+        address l2ExchangeRateProvider,
+        address rateLimiter,
+        address tokenOut,
+        uint32 dstEid,
+        address messenger,
+        address receiver,
+        address delegate
+    ) external virtual initializer {
+        
```

Comment:
we can remove this line so that is has the same # of LOC with that of other syncpool contracts